java/src/jmri/Manager.java

```diff
@@ -199,8 +195,9 @@ public default String validateSystemNameFormat(@Nonnull String name) throws BadS
      * Overriding implementations may rely on
      * {@link #validSystemNameFormat(java.lang.String)}, however they must
      * provide an actionable message in the thrown exception if that method does
-     * not return {@link NameValidity#VALID}. Implementations of
-     * this method <em>must not</em> throw an exception, log an error, or
+     * not return {@link NameValidity#VALID}. When overriding implementations
+     * of this method rely on validSystemNameFormat(), implementations of
```

Comment:
Why talk about the required behavior of validSystemNameFormat() here, when it's already covered in the Javadoc for that method?

Reply:
Prior to 4.17.2, implementations of `validSystemNameFormat()` did present dialogs, throw runtime exceptions, and log errors about the validity of the system name. It was a bit of a mess.

I'm not claiming that those implementations are _wrong_, just that if they do that, they can't be relied upon within this method, since spamming a log with error messages or interrupting a user typing a name won't be appreciated by users.